src/DDTrace/Integrations/SymfonyMessenger/SymfonyMessengerIntegration.php

```diff
@@ -0,0 +1,348 @@
+            // Only exists in Symfony Messenger 4.4+
+            return Integration::NOT_LOADED;
+        }
+
+        trace_method(
+            'Symfony\Component\Messenger\MessageBusInterface',
+            'dispatch',
+            function (SpanData $span, array $args, $retval, $exception) use ($integration) {
+                $integration->setSpanAttributes($span, 'symfony.messenger.dispatch', null, $retval ?? $args[0], null, null, true);
+                if ($exception) {
+                    // Worker::handleMessage() will catch the exception. We need to manually attach it to the root span.
+                    \DDTrace\root_span()->exception = $exception;
+                }
+            }
+        );
+
+        // Attach current context to Envelope before sender sends it to remote queue
+        install_hook(
+            'Symfony\Component\Messenger\Transport\Sender\SenderInterface::send',
+            function (HookData $hook) {
+                /** @var \Symfony\Component\Messenger\Envelope $envelope */
+                $envelope = $hook->args[0];
+
+                if (\ddtrace_config_distributed_tracing_enabled()) {
+                    $ddTraceStamp = $envelope->last(DDTraceStamp::class);
+
+                    // Add distributed tracing stamp only if not already on the envelope
+                    if ($ddTraceStamp === null) {
+                        $tracingHeaders = \DDTrace\generate_distributed_tracing_headers();
+                        $hook->overrideArguments([
+                            $envelope->with(new DDTraceStamp($tracingHeaders))
+                        ]);
+                    }
+                }
+            }
+        );
+
+        trace_method(
+           'Symfony\Component\Messenger\Transport\TransportInterface',
+           'send',
+           function (SpanData $span, array $args, $envelope) use ($integration) {
+               $integration->setSpanAttributes($span, 'symfony.messenger.send', null, $envelope ?? $args[0], null, 'send', true);
+           }
+        );
+
+        trace_method(
+            'Symfony\Component\Messenger\Worker',
+            'handleMessage',
+            [
+                'prehook' => function (SpanData $span, array $args) use ($integration) {
+                    /** @var \Symfony\Component\Messenger\Envelope $envelope */
+                    $envelope = $args[0];
+                    /** @var string|ReceiverInterface $transportName */
+                    $transportName = $args[1];
+                    if (\is_object($transportName)) {
+                        $transportName = \get_class($transportName);
+                    }
+
+                    $integration->setSpanAttributes(
+                        $span,
+                        'symfony.messenger.consume',
+                        null,
+                        $envelope,
+                        $transportName,
+                        'receive'
+                    );
```

Comment:
If we set `DD_TRACE_REMOVE_ROOT_SPAN_SYMFONY_MESSENGER=true`, would this be the root span, and hence appear as a resource? That would be my desired outcome, so that message handlers appear alike to controllers for HTTP traffic.

Reply:
Yes, that's indeed the goal, and also one of the reasons this env var will be enabled by default :+1: If the root span isn't removed, it will most likely be the `messenger:consume` command span; however, this operation will certainly be a long-running job, consuming multiple messages.